test/e2e/upgrades/topology_manager.go

```diff
@@ -0,0 +1,317 @@
+		if !ok {
+			return fmt.Errorf("Unknown sibilings for cpu %d", cpuNumPrev)
+		}
+
+		cpuNum := allowed[i]
+		sibilings, ok := cpuMap[fmt.Sprintf("SIBILING%d", cpuNum)]
+		if !ok {
+			return fmt.Errorf("Unknown sibilings for cpu %d", cpuNum)
+		}
+
+		if !reflect.DeepEqual(sibilingsPrev, sibilings) {
+			return fmt.Errorf("disaligned cpus %d and %d (%v and %v)", cpuNumPrev, cpuNum, sibilingsPrev, sibilings)
+		}
+	}
+	return nil
+}
+
+func isRelevantUpgrade(upgrade UpgradeType) bool {
+	return upgrade == NodeUpgrade || upgrade == ClusterUpgrade
+}
+
+func getTopologyManagerSingleNumaNodePolicyEnabledNode(f *framework.Framework) *v1.Node {
+	selector := labels.Set{"node-role.kubernetes.io/worker=": ""}.AsSelector()
+	nodeList, err := f.ClientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: selector.String(),
+	})
+	framework.ExpectNoError(err)
+
+	ginkgo.By("Finding a worker node with Topology Manager configured")
+
+	for _, node := range nodeList.Items {
+		kubeletConfig, err := getCurrentKubeletConfig(node.ObjectMeta.Name, f.Namespace.Name)
+		framework.ExpectNoError(err)
+
+		framework.Logf("node %q TopologyManagerPolicy %q", node.ObjectMeta.Name, kubeletConfig.TopologyManagerPolicy)
+		if kubeletConfig.TopologyManagerPolicy == topologymanager.PolicySingleNumaNode {
+			return &node
+		}
+	}
+
+	return nil
+}
+
+// TODO: this dupes test/e2e/windows/memory_limits.go. How can we factor this code out?
+// getCurrentKubeletConfig modified from test/e2e_node/util.go
+func getCurrentKubeletConfig(nodeName, namespace string) (*kubeletconfig.KubeletConfiguration, error) {
+
+	resp := pollConfigz(5*time.Minute, 5*time.Second, nodeName, namespace)
+	kubeCfg, err := decodeConfigz(resp)
+	if err != nil {
+		return nil, err
+	}
+	return kubeCfg, nil
+}
+
+// Causes the test to fail, or returns a status 200 response from the /configz endpoint
+func pollConfigz(timeout time.Duration, pollInterval time.Duration, nodeName, namespace string) *http.Response {
+	// start local proxy, so we can send graceful deletion over query string, rather than body parameter
+	ginkgo.By("Opening proxy to cluster")
+	tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, namespace)
+	cmd := tk.KubectlCmd("proxy", "-p", "0")
+	stdout, stderr, err := framework.StartCmdAndStreamOutput(cmd)
+	framework.ExpectNoError(err)
+	defer stdout.Close()
+	defer stderr.Close()
+	defer framework.TryKill(cmd)
+	buf := make([]byte, 128)
+	var n int
+	n, err = stdout.Read(buf)
+	framework.ExpectNoError(err)
+	output := string(buf[:n])
+	proxyRegexp := regexp.MustCompile("Starting to serve on 127.0.0.1:([0-9]+)")
+	match := proxyRegexp.FindStringSubmatch(output)
+	framework.ExpectEqual(len(match), 2)
+	port, err := strconv.Atoi(match[1])
+	framework.ExpectNoError(err)
+	ginkgo.By("http requesting node kubelet /configz")
```

Comment:
nit: better to put a new blank line before `ginkgo.By()` for code readability.

Reply:
done